docs-configuration.md

```diff
@@ -0,0 +1,352 @@
+                   'my_resource': '{my_project}_some_resource'
+                }).add_configuration(
+                name='prod',
+                properties={
+                   'my_project': 'prod-project-id'
+                })
+
+config.pretty_print('dev')
+config.pretty_print('prod')
+```  
+
+output:
+
+```text
+dev config:
+{   'my_project': 'dev-project-id',
+    'my_resource': 'dev-project-id_some_resource'}
+prod config:
+{   'my_project': 'prod-project-id',
+    'my_resource': 'prod-project-id_some_resource'}
+
+```
+
+There is a **special placeholder**: `env` for properties containing environment name. For example,
+the `gcp_project_id` property is resolved from:
+
+```python
+'gcp_project_id': '{env}-project-id'
+```
+
+to `dev-project-id` on `dev` and to `prod-project-id` on `prod`.
+
+### Master configuration
+
+In a `Config` object you can define a master configuration. 
+Any property defined in the master configuration is *inherited* by other configurations.  
+
+By default, the configuration defined in the `Config` init method is the master one:
+
+
+```python
+config = Config(name='dev',
+                properties={
+                   'my_project': 'my_value'
+                }).add_configuration(
+                name='prod',
+                properties={})
+
+config.pretty_print('dev')
+config.pretty_print('prod')
+```
+
+output:
+
+```text
+dev config:
+{'my_project': 'my_value'}
+prod config:
+{'my_project': 'my_value'}
+```
+
+You can disable properties inheritance by setting the `is_master` flag to `False`:
+
+```python
+config = Config(name='dev',
+                is_master=False,
+                properties={
+                   'my_project': 'my_value'
+                }).add_configuration(
+                name='prod',
+                properties={})
+
+config.pretty_print('dev')
+config.pretty_print('prod')
+```
+
+output:
+
+```text
+dev config:
+{'my_project': 'my_value'}
+prod config:
+{}                
+```                
+
+### Default configuration
+
+A default configuration is used when no environment name is given. 
+By default, the configuration defined in the `Config` init method is the default one.
+It is chosen while resolving properties, when `env_name` is `None`. For example:
+
+
+```python
+config = Config(name='dev',
+                properties={
+                   'my_project': 'I_am_{env}'
+                }).add_configuration(
+                name='prod',
+                properties={})
+
+config.pretty_print()
+config.pretty_print('dev')
+
+```
+
+output:
+
+```text
+dev config:
+{'my_project': 'I_am_dev'}
+dev config:
+{'my_project': 'I_am_dev'}
+```
+
+You can change the roles by setting the `is_default` flag:
+
+```python
+config = Config(name='dev',
+                is_default=False,
+                properties={
+                   'my_project': 'I_am_{env}'
+                }).add_configuration(
+                name='prod',
+                is_default=True,
```

Comment:
what happens when:
- each configuration sets a flag value false?
- only first configuration sets a flag value false, second one doesn't specify it at all?

Reply:
defaults are reasonable, by default in init method `is_default=True` and in add_configuration `is_default=False`. If you set all to False